src/core/background/scrobbler/listenbrainz-scrobbler.js

```diff
@@ -284,6 +284,14 @@ define((require) => {
 				trackMeta.additional_info.release_artist_name = song.getAlbumArtist();
 			}
 
+			if (song.getUniqueId() && song.metadata.label === 'Spotify') {
+				trackMeta.additional_info.spotify_id = song.getUniqueId();
+			}
+
+			if (song.getDuration()) {
+				trackMeta.additional_info.duration = song.getDuration();
```

Comment:
We now [clarified](https://listenbrainz.readthedocs.io/en/latest/users/json.html#payload-json-details) that we accept either `duration` or `duration_ms`, the former for durations in seconds and the latter (which I would recommend) for durations in milliseconds

Reply:
What's the rational for using `ms`? internally the connector uses seconds for duration so we would have the same precision if we multiplied this value up?

Reply:
In that case `duration` is the correct one, sorry if I wasn't clear.
I just wasn't sure of the internal implementation and wanted to make sure the right field is used.